pkg/controllers/user/dnsrecord/dnsrecord.go

```diff
@@ -97,6 +110,18 @@ func (c *Controller) reconcileEndpoints(key string, obj *corev1.Service) error {
 		service := namespaceService[1]
 		targetEndpoint, err := c.endpointLister.Get(namespace, service)
 		if err != nil {
+			exists, name, deleting := c.checkLinkedSvc(namespace, service, obj.Spec.ExternalName)
```

Comment:
isn't exists an opposite to deleting? deleting should be treated as non existent

Reply:
Treating it as non existent, just logging the msg and below part would delete the alias